### ISSUE TITLE
Just AWESOME !!!

### ISSUE THREAD

**Issue by nueko** [pshrmn/curi#1]
Thanks for this package, I just start to use react to my current project and it's awesome to know this project.

Thanks alot.

**Comment by pshrmn** [comment]
Thank you. This project is still evolving a bit (I'm currently writing a new history implementation), but I really like where it is at.

**Comment by bboydflo** [comment]
Just few quick questions.

1. What was the main motivation for this package? I'm wondering why this package exists. What is missing in other packages? 

2. Can this be used with Preact?

BTW, thanks for the docs, they are awesome!

**Comment by pshrmn** [comment]
@bboydflo 

1. I was helping with React Router and there were some issues that kept cropping up that had me wondering how I would do things differently. Originally I was trying to think of how to improve RR, but at a certain point I realized that to do what I wanted, I had to start from scratch. Once I wrote the basic router, I realized that this didn't have to just be limited to React and that really expanded the scope of the project.

2. That is a good question. The `@curi/react` components _should_ work with Preact, but I haven't tested it; I'll have to look into that. If they do not, I could probably either Preact-proof the `@curi/react` package or just create a `@curi/preact` package.

3. Glad you like the docs. I tried to make them cover everything someone needs to know to get up to speed quickly. Now I just need to figure out how to get people to find out about Curi :sweat_smile:.
